Source/WebCore/crypto/mac/CryptoKeyOKPCocoa.cpp

```diff
@@ -44,28 +44,33 @@ bool CryptoKeyOKP::isPlatformSupportedCurve(NamedCurve namedCurve)
 
 std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmIdentifier identifier, NamedCurve namedCurve, bool extractable, CryptoKeyUsageBitmap usages)
 {
-    if (namedCurve != NamedCurve::Ed25519 && namedCurve != NamedCurve::X25519)
+    if (!isPlatformSupportedCurve(namedCurve))
         return { };
 
     ccec25519pubkey ccPublicKey;
     ccec25519secretkey ccPrivateKey;
-
-    if (identifier == CryptoAlgorithmIdentifier::Ed25519) {
-        int ccerror = 0;
+    int ccerror = 0;
```

Comment:
Nit: `ccerror` doesn't really do anything outside of the two scopes, so it could be simply declared and used in both curve-specific scopes, IMO.

Reply:
Done